Sources/BraveWallet/Crypto/Stores/NFTStore.swift

```diff
@@ -303,42 +303,44 @@ public class NFTStore: ObservableObject, WalletObserverStore {
       for networkAssets in [userVisibleNFTs, userHiddenNFTs, unionedSpamNFTs] {
         allNFTs.append(contentsOf: networkAssets.flatMap(\.tokens))
       }
-      // fetch balance for all NFTs
-      let allAccounts = filters.accounts.map(\.model)
-      nftBalancesCache = await withTaskGroup(
-        of: [String: [String: Int]].self,
-        body: { @MainActor [nftBalancesCache, rpcService] group in
-          for nft in allNFTs { // for each NFT
-            guard let networkForNFT = allNetworks.first(where: { $0.chainId == nft.chainId }) else {
-              continue
-            }
-            group.addTask { @MainActor in
-              let updatedBalances = await withTaskGroup(
-                of: [String: Int].self,
-                body: { @MainActor group in
-                  for account in allAccounts where account.coin == nft.coin {
-                    group.addTask { @MainActor in
-                      let balanceForToken = await rpcService.balance(
-                        for: nft,
-                        in: account,
-                        network: networkForNFT
-                      )
-                      return [account.address: Int(balanceForToken ?? 0)]
+      // if we're not hiding unowned or grouping by account, balance isn't needed
+      if filters.isHidingUnownedNFTs {
```

Comment:
nit: I think we also want to include `|| filters.groupBy == .accounts`, in case we later remove logic that sets `isHidingUnownedNFTs` to true when grouping my accounts? Presumably would get caught in that PR though.